### DISCUSSION TITLE
Error when attempting to run otioview for the first time (Mac M1)

### DISCUSSION THREAD

**Discussion by TrevorAyl** [AcademySoftwareFoundation/OpenTimelineIO#1671] [Q&A] [answered]
Apologies if this is Python 101 but I'm getting the following error when I attempt to run otioview...
`IndexError: Signature "triggered()" not found for signal: "triggered".`

Fuller picture... I have uninstalled and reinstalled packages a couple of times... (because I subsequently get `zsh: operation not permitted: otioview` when trying to run same command)

`*** ~ % pip install PySide6 
Collecting PySide6
  Obtaining dependency information for PySide6 from https://files.pythonhosted.org/packages/54/e9/8d6e64aecc19a8e3ca899f79ab9bbd831c2e4cfaf1ac89e084c9c5bd8e39/PySide6-6.6.0-cp38-abi3-macosx_11_0_universal2.whl.metadata
  Using cached PySide6-6.6.0-cp38-abi3-macosx_11_0_universal2.whl.metadata (5.1 kB)
Requirement already satisfied: shiboken6==6.6.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from PySide6) (6.6.0)
Requirement already satisfied: PySide6-Essentials==6.6.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from PySide6) (6.6.0)
Requirement already satisfied: PySide6-Addons==6.6.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from PySide6) (6.6.0)
Using cached PySide6-6.6.0-cp38-abi3-macosx_11_0_universal2.whl (6.7 kB)
Installing collected packages: PySide6
Successfully installed PySide6-6.6.0
***  ~ % pip install opentimelineio        
Collecting opentimelineio
  Using cached OpenTimelineIO-0.15.0-cp311-cp311-macosx_12_0_universal2.whl
Requirement already satisfied: pyaaf2<1.7,>=1.4 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from opentimelineio) (1.6.0)
Installing collected packages: opentimelineio
Successfully installed opentimelineio-0.15.0
***  ~ % pip install opentimelineio-plugins
Collecting opentimelineio-plugins
  Obtaining dependency information for opentimelineio-plugins from https://files.pythonhosted.org/packages/b3/21/5c0ed55f3680eecd048a25a74ada57db8ef54e73f6fae45044ff479e1f2e/opentimelineio_plugins-0.15.0-py3-none-any.whl.metadata
  Using cached opentimelineio_plugins-0.15.0-py3-none-any.whl.metadata (15 kB)
Requirement already satisfied: opentimelineio==0.15.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from opentimelineio-plugins) (0.15.0)
Requirement already satisfied: pyaaf2<1.7,>=1.4 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from opentimelineio==0.15.0->opentimelineio-plugins) (1.6.0)
Using cached opentimelineio_plugins-0.15.0-py3-none-any.whl (11 kB)
Installing collected packages: opentimelineio-plugins
Successfully installed opentimelineio-plugins-0.15.0
***  ~ % otioview /Users/***/Desktop/test.edl 
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/bin/otioview", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/opentimelineview/console.py", line 292, in main
    window = Main(
             ^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/opentimelineview/console.py", line 153, in __init__
    self._create_navigation_menu(navigation_menu)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/opentimelineview/console.py", line 222, in _create_navigation_menu
    navigation_menu.triggered[[QAction]].connect(__callback)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
IndexError: Signature "triggered()" not found for signal: "triggered". Available candidates: "triggered(QAction*)"`

**Comment by TrevorAyl** [comment]
Here's chatgpt3.5 answer...

The error message you're encountering is related to a compatibility issue with the PySide version used by OpenTimelineIO (OTIO) and its dependencies. It seems like there might be a conflict in the Qt libraries.

To resolve this issue, you can try the following steps:

1. **Check Your Python Environment**:
   Ensure that you're using the same Python environment for both PySide and OpenTimelineIO. The error can occur when there are multiple Python environments or installations.

2. **Update or Reinstall PySide**:
   The error message you provided shows that you have PySide6 installed, which is compatible with Python 3.8. If you are using Python 3.11, it might be causing issues. You should install PySide6 compatible with Python 3.11. You can do this by running:

   ```
   pip install PySide6==6.6.0
   ```

   Make sure it's the version of PySide6 that matches your Python version.

3. **Verify Dependencies**:
   Confirm that you have the correct versions of all dependencies installed. You can check OpenTimelineIO's official documentation or release notes to see if any specific versions are recommended.

4. **Check Qt Compatibility**:
   Ensure that the version of Qt used by PySide is compatible with the version of Qt used by OpenTimelineIO. If they are mismatched, it can cause issues. You may need to install the correct version of Qt or PyQt for your Python environment.

5. **Use a Virtual Environment**:
   Consider creating a Python virtual environment to isolate your project's dependencies and avoid conflicts with other Python packages.

6. **Contact the OpenTimelineIO Community**:
   If the issue persists, you can seek help from the OpenTimelineIO community. They may have encountered similar issues and could provide guidance or updates related to compatibility.

Keep in mind that the error message indicates a compatibility issue with PySide, so ensuring that all components are compatible and properly aligned in your Python environment is crucial to resolve this error.

**Reply by richardssam** [reply, marked as answer]
You can edit "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/opentimelineview/console.py" and change line 222 from:    
`   navigation_menu.triggered[[QAction]].connect(__callback)
`       
to
`   navigation_menu.triggered.connect(__callback)
`

And it will at least launch for me. Haven't spent enough time looking at it to know if that should be a permanent fix.
